src/stan/mcmc/stepsize_adaptation.hpp

```diff
@@ -70,7 +70,10 @@ class stepsize_adaptation : public base_adaptation {
     epsilon = std::exp(x);
   }
 
-  void complete_adaptation(double& epsilon) { epsilon = std::exp(x_bar_); }
+  void complete_adaptation(double& epsilon) {
+    if (counter_ > 0)
+      epsilon = std::exp(x_bar_);
```

Comment:
This will generate a little nicer code with a cmove instead of a branch
```suggestion
      epsilon = (counter_ > 0) ? std::exp(x_bar_) : epsilon;
```

Reply:
This code gets called exactly once per chain, so I'd actually vote for the more obvious conditional